jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -692,7 +692,7 @@ static void write(Sink sink, boolean last, ByteBuffer byteBuffer) throws IOExcep
          */
         static void write(Sink sink, boolean last, String utf8Content, Callback callback)
         {
-            sink.write(last, StandardCharsets.UTF_8.encode(utf8Content), callback);
+            sink.write(last, ByteBuffer.wrap(utf8Content.getBytes(StandardCharsets.UTF_8)), callback);
```

Comment:
Feels like a lot of data copying and allocation, and something that may even depend on the Java version.

What Java versions have you tried? It is the case that the performance is that bad in all recent Java versions?

Reply:
I've tried both JDK 17 and 22 and they have similar performance, which can be summed up as following:

 - when the string is pure ASCII, wrap/getBytes just a memcopy and is ridiculously faster than encode, about 20X faster! Profiling the allocation rate shows that mem bandwidth is the bottleneck.
 - the more you move away from pure ASCII, the lesser the perf benefits become: about 2.5X faster for FR, about 50% faster for JA. Allocations also increase: 5X for FR, 20% for JA.